openedx/tests/xblock_integration/test_done.py

```diff
@@ -0,0 +1,233 @@
+"""
+This tests that the completion XBlock correctly stores state. This
+is a fairly simple XBlock, and a correspondingly simple test suite.
+"""
+
+import json
+import mock
+import unittest
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+
+from lms.djangoapps.courseware.tests.helpers import LoginEnrollmentTestCase
+from lms.djangoapps.courseware.tests.factories import GlobalStaffFactory
+
+
+class SimpleEventTestMixin(object):
```

Comment:
I definitely plan on doing this, but I would prefer to do so in a week or two, either with the XBlock testing PR or the RateXBlock PR. I am talking to @benpatterson about how we want to structure XBlocks tests, and we'll also have a better idea of whether this is properly generic once it is working on two XBlocks. Before then, I'd rather not make this a common utility. 

If I need it for the :+1:, I'm okay doing it now, but I think that'd be a more logical time to do so.

I started with EventTestMixin, but it didn't really contribute much. It just added a thin layer of indirection between itself and a tracker (where most of the functionality lived). It added a fair bit of complexity, wasn't really documented, and didn't save on any code. The examples of how it was used were a little bit messy. Ben suggested looking at the cohorts one, which was a lot more developed and fairly nice (it puts events in Mongo, and you search from them), but Ben said that it requires bok_choy. When I played with moving this into bok_choy, testing got very slow.

(The docstring was intended to convey this)